platform-sdk/swirlds-benchmarks/src/jmh/java/com/swirlds/benchmark/VirtualMapBaseBench.java

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.benchmark;
+
+import static com.swirlds.common.threading.manager.AdHocThreadManager.getStaticThreadManager;
+
+import com.swirlds.common.io.streams.SerializableDataInputStream;
+import com.swirlds.common.io.streams.SerializableDataOutputStream;
+import com.swirlds.common.threading.framework.config.ThreadConfiguration;
+import com.swirlds.virtualmap.VirtualMap;
+import com.swirlds.virtualmap.internal.merkle.VirtualRootNode;
+import com.swirlds.virtualmap.internal.pipeline.VirtualRoot;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.IntStream;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.openjdk.jmh.annotations.TearDown;
+
+public abstract class VirtualMapBaseBench extends BaseBench {
```

Comment:
All these fields/methods can be kept in `VirtualMapBench`, no hard to have them there and inherit in `ReconnectBench`. This is exactly what happens with `CryptoBench`: it extends from `VirtualMapBench` and adds a few more benchmarks

Reply:
Unless a specific benchmark method name is requested when running JMH, this approach leads to running all the methods annotated with `@Benchmark`, including those defined in base classes. So I refactored this code on purpose to avoid inheriting from a class that has other benchmarks defined already, so as to simplify launching JMH.